### DISCUSSION TITLE
[FIDI] Mail report improvement

### DISCUSSION THREAD

**Discussion by MacPaille** [firefly-iii/firefly-iii#5641] [General]
Hey JC5,

I have configured the mail report on my docker instance that runs in cli mode every day automatically at 7PM. I ran it to test and I see I'm getting 1 email per account with imported transactions.
As such it's not an issue except I'm getting 10 mails... But the thing I find missing is the name of the account in the email. I get the list of transactions that are imported but not for which account...

Would it be possible to add it?
Maybe an optimisation would be to summarize all the reports into 1 email and send it only once.

**Comment by JC5** [comment]
So you have 10 separate JSON files, is that right?

**Reply by MacPaille** [reply]
Yes one per account. Isn't how it's suppose to be? Did I miss a hidden feature? 😄 

I mean each run it parses all the json and import either with Nordigen or with a CSV. I'm expecting a mail summary at the end, not for each import :)

**Reply by JC5** [reply]
Each import is completely separate, it doesn't remember a state. I can't change that unfortunately.

**Reply by MacPaille** [reply]
OK no problem I'll handle the 10 mails. But would it be possible to add the name of the account it has been imported to?

**Reply by JC5** [reply]
I'll write it down, yes!

**Reply by MacPaille** [reply]
Maybe another improvement to reduce the number of emails. Only send them when there is a useful message like and error ro real transactions added. Most of my runs are using same CSV data so nothing is added. I don't need a mail for that.

Example of mail (sorry can't paste an image, it's restricted on my pc):

Result of your import on 2022-01-26 @ 10:09:13
Errors
Warnings
Messages
Line #1: Converted CSV line 1 into a transaction.

So I'm getting it only because it's converting a line from my CSV but that does not result into a transaction as it's a duplicate.

**Reply by JC5** [reply]
https://github.com/firefly-iii/data-importer/blob/main/.env.example#L63

**Reply by MacPaille** [reply]
That is my point I have already set IGNORE_DUPLICATE_ERRORS to true. Meaning I dont get the mail because of the duplicates I'm getting it due to this line:
Line #1: Converted CSV line 1 into a transaction.

Which I totally don't care about :) It's purely something internal to the FIDI tool showing in the logs it has converted a CSV, but as it does not result in any transactions it should just not be part of the email report. At least in my point of view.

**Reply by JC5** [reply]
Ahhh I see it, I'll check it out. That doesn't have to be a message.

**Reply by MacPaille** [reply]
Indeed 😄  having that filtered out I should only get emails with new transactions created or errors/warnings. Which is all I need 😄

**Reply by JC5** [reply]
0.7.1 should no longer show the extra lines. Unfortunately I can't easily import account information into the message.

**Reply by MacPaille** [reply]
I tried to pull the latest develop docker image but it still starts as v0.7.0. Is there something missing?

**Reply by JC5** [reply]
I ran the build again, let me know.

**Reply by MacPaille** [reply]
Got it. I'm still getting the email for each json file, this time juste completely empty :)

![image](https://user-images.githubusercontent.com/31923233/151300089-1c185140-caa7-4148-b674-a55eca5281cd.png)

**Reply by JC5** [reply]
Weird, I thought I caught that ages ago.

**Reply by JC5** [reply]
Built the same version again, if you get empty messages send me the logs.

**Reply by MacPaille** [reply]
I meant I got the 0.7.1 and now the csv converted lines do not appear. But now I just an empty email for each json.

Buy empty I mean the skeleton like on the screenshot above but nothing in each of the 3 categories.

**Reply by JC5** [reply]
I know what you mean, the screenshot is obvious 😉 

There is a pretty basic check in the code, so I'm just surprised. I added some logging to the mail routine.

**Reply by MacPaille** [reply]
@JC5 I think there is an issue with latest develop image with the fix. Getting this error after importing the transactions from my account, I suppose at the time of sending the mail.

[2022-01-27 19:02:28] production.INFO: Done submitting 23 transactions to your Firefly III instance.  
[2022-01-27 19:02:28] production.DEBUG: Now in setSubmissionStatus(submission_done)  
[2022-01-27 19:02:28] production.DEBUG: Found "conv-4thA95ekyxLW" in the session  
[2022-01-27 19:02:28] production.DEBUG: Now in startOrFindJob(conv-4thA95ekyxLW)  
[2022-01-27 19:02:28] production.DEBUG: Try to see if file exists for job conv-4thA95ekyxLW.  
[2022-01-27 19:02:28] production.DEBUG: Status file exists for job conv-4thA95ekyxLW.  
[2022-01-27 19:02:28] production.DEBUG: Now in App\Services\Shared\Import\Status\SubmissionStatusManager::storeSubmissionStatus(conv-4thA95ekyxLW): submission_done  
[2022-01-27 19:02:28] production.DEBUG: Created event ImportedTransactions with filtering  
[2022-01-27 19:02:28] production.ERROR: Array to string conversion {"exception":"[object] (ErrorException(code: 0): Array to string conversion at /var/www/html/app/Events/ImportedTransactions.php:54)
[stacktrace]
#0 /var/www/html/app/Events/ImportedTransactions.php(54): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1 /var/www/html/app/Console/AutoImports.php(203): App\\Events\\ImportedTransactions->__construct()
#2 /var/www/html/app/Console/AutoImports.php(153): App\\Console\\Commands\\AutoImport->importFile()
#3 /var/www/html/app/Console/Commands/AutoImport.php(85): App\\Console\\Commands\\AutoImport->importFiles()
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\\Console\\Commands\\AutoImport->handle()
#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Util.php(40): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(653): Illuminate\\Container\\BoundMethod::call()
#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(136): Illuminate\\Container\\Container->call()
#10 /var/www/html/vendor/symfony/console/Command/Command.php(298): Illuminate\\Console\\Command->execute()
#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\\Component\\Console\\Command\\Command->run()
#12 /var/www/html/vendor/symfony/console/Application.php(1005): Illuminate\\Console\\Command->run()
#13 /var/www/html/vendor/symfony/console/Application.php(299): Symfony\\Component\\Console\\Application->doRunCommand()
#14 /var/www/html/vendor/symfony/console/Application.php(171): Symfony\\Component\\Console\\Application->doRun()
#15 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Application.php(94): Symfony\\Component\\Console\\Application->run()
#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(129): Illuminate\\Console\\Application->run()
#17 /var/www/html/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle()
#18 {main}
"}

**Reply by JC5** [reply]
Oh thanks, I built nr 86, could you check it out?

**Reply by MacPaille** [reply]
Works fine as expected :) I got only emails to accounts with new transactions. Thanks a lot!

Just to know, if the Bank agreement with the Nordigen sessions expires or sends an error (some banks limit the number of polling to the account) will it appear in a mail?

**Reply by JC5** [reply]
Errors are generally not reported over mail (yet), the mail message is purely for the import.

I have to work on making these things more robust, it's on my list to do so!

**Reply by MacPaille** [reply]
Thanks a lot for all the work! The more I use it the more I find it awesome. I'm almost not doing anything manually anymore to import my transactions every day.
